tekton-resources/tasks/run-wc-e2e-tests.yaml

```
@@ -26,6 +26,54 @@ spec:
     description: "The test suites to run depending on the repo that triggered the run"
     default: "."
 
+  - name: VPN_ENABLED
+    type: string
+    description: "If set to true  test suites to run depending on the repo that triggered the run"
+    default: "false"
+      
+  sidecars:
+    - name: vpn
+      image: kylemanna/openvpn:2.4
+      env:
+        - name: "VPN_ENABLED"
+          value: "$(params.VPN_ENABLED)"
+      securityContext:
+        capabilities:
+          add:
+          - NET_ADMIN
+      script: |
+        #!/usr/bin/env bash
+        set -e
+        if [ "${VPN_ENABLED}" == "true" ]; then
+            echo ">> Starting VPN"
+            echo
+            openvpn --config /openvpn/ci.ovpn &
+            until nslookup google.com 10.0.253.1
+            do
+              echo "waiting until VPN is ready"
+              sleep 2s
+            done
+            touch /shared/vpn-ready
+
+            # wait until the test finished
+            until [ -f "/shared/run-test-finished" ]
+            do
+              echo "waiting until test are finished"
+              sleep 1s
+            done
+        else
+            echo "VPN is not enabled, VPN_ENABLED=${VPN_ENABLED}"
+        fi
```

```suggestion
        if [ "${VPN_ENABLED}" == "true" ]; then
            echo ">> Starting VPN"
            echo
            openvpn --config /openvpn/ci.ovpn &
            until nslookup google.com 10.0.253.1
            do
              echo "waiting until VPN is ready"
              sleep 2s
            done
        else
            echo "VPN is not enabled, VPN_ENABLED=${VPN_ENABLED}"
        fi
        
        touch /shared/vpn-ready

        # wait until the test finished
        until [ -f "/shared/run-test-finished" ]
        do
          echo "waiting until test are finished"
          sleep 1s
        done
```

If we move the done and check logic out of the if statement we can use it for both.

---

👍